code generators/de.fhdo.lemma.model_processing.code_generation.container_base/src/main/java/de/fhdo/lemma/model_processing/code_generation/container_base/file/property/OpendPropertyFiles.xtend

```diff
@@ -0,0 +1,101 @@
+package de.fhdo.lemma.model_processing.code_generation.container_base.file.property
+
+import java.util.HashSet
+import java.util.Set
+import java.io.File
+
+/**
+ * This class manages several different application.properties files, used in the spring framework.
+ * The class is implemented by the usage of the singleton design pattern.
+ *
+ *
+ * @author [Philip Wizenty](mailto:philip.wizenty@fh-dortmund.de)
+ */
+class OpendPropertyFiles {
+    static OpendPropertyFiles instance
+    val Set<PropertyFile> propertyFiles = new HashSet
+
+    static def OpendPropertyFiles getInstance() {
+        if (OpendPropertyFiles.instance === null)
+            OpendPropertyFiles.instance = new OpendPropertyFiles
+
+        return OpendPropertyFiles.instance
+    }
+
+    /*
+     * Add the given [propertyFile] to the store or replace it, if it already exists
+     */
+    def addOrReplace(PropertyFile propertyFile) {
+        propertyFiles.removeIf(p | p.filePath == propertyFile.filePath)
```

Comment:
Err, please add an `equals` and `hashCode` implementation to the `PropertyFile` class based on the `filePath`, because `Set` uses those to automatically check objects for equality. Something like this should suffice:
```
override hashCode() {
    return filePath.hashCode()
}

override equals(Object object) {
    if (object === null || !(object instanceof PropertyFile))
        return false

    return filePath == object.filePath
}

Reply:
Another question: Why is it even necessary at all to replace the PropertyFile instance in the set?